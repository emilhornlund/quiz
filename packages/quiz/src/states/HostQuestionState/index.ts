export type { HostQuestionStateProps } from './HostQuestionState'
export { default } from './HostQuestionState'
