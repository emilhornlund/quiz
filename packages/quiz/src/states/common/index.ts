export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export type { QuestionAnswerPickerProps } from './QuestionAnswerPicker'
export type { QuestionMediaProps } from './QuestionMedia'
export type { QuestionTextPreviewProps } from './QuestionTextPreview'
export type { QuestionTypePointsBarProps } from './QuestionTypePointsBar'

export { default as GamePage } from './GamePage'
export { default as NonInteractiveInfoBox } from './NonInteractiveInfoBox'
export { default as PointsBehindIndicator } from './PointsBehindIndicator'
export { default as QuestionAnswerPicker } from './QuestionAnswerPicker'
export { default as QuestionMedia } from './QuestionMedia'
export { default as QuestionTextPreview } from './QuestionTextPreview'
export { default as QuestionTypePointsBar } from './QuestionTypePointsBar'
