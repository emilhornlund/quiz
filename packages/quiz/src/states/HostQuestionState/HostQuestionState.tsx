import { faUserGroup } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { GameQuestionHostEvent, MediaType } from '@quiz/common'
import React, { FC } from 'react'

import {
  HostGameFooter,
  IconButtonArrowRight,
  Page,
  ProgressBar,
  QuestionAnswerPicker,
  Typography,
} from '../../components'
import ResponsiveImage from '../../components/ResponsiveImage'
import ResponsivePlayer from '../../components/ResponsivePlayer'
import { classNames } from '../../utils/helpers.ts'

import styles from './HostQuestionState.module.scss'

export interface HostQuestionStateProps {
  event: GameQuestionHostEvent
  onSkip: () => void
}

const HostQuestionState: FC<HostQuestionStateProps> = ({
  event: {
    game: { pin },
    question,
    countdown,
    submissions: { current: currentSubmission, total: totalSubmissions },
    pagination: { current: currentQuestion, total: totalQuestions },
  },
  onSkip,
}) => (
  <Page
    height="full"
    align="space-between"
    header={
      <IconButtonArrowRight
        id={'skip-button'}
        type="button"
        kind="secondary"
        size="small"
        value="Skip"
        onClick={onSkip}
      />
    }
    footer={
      <HostGameFooter
        gamePIN={pin}
        currentQuestion={currentQuestion}
        totalQuestions={totalQuestions}
      />
    }>
    <div className={classNames(styles.row, styles.flexibleHeight)}>
      <div className={styles.column}>
        <Typography variant="subtitle">{question.question}</Typography>
      </div>
    </div>
    <div className={classNames(styles.row, styles.flexibleHeight)}>
      <div className={styles.column}>
        <div className={classNames(styles.iconInfo)}>
          <FontAwesomeIcon icon={faUserGroup} />
          <span>
            {currentSubmission} / {totalSubmissions}
          </span>
        </div>
      </div>
    </div>
    <div
      className={classNames(styles.row, styles.fullHeight, styles.fullWidth)}>
      {question.media?.type === MediaType.Image && (
        <ResponsiveImage
          imageURL={question.media.url}
          alt={question.question}
        />
      )}
      {question.media?.type === MediaType.Audio ||
        (question.media?.type === MediaType.Video && (
          <ResponsivePlayer url={question.media.url} />
        ))}
    </div>
    <div
      className={classNames(
        styles.row,
        styles.fullWidth,
        styles.flexibleHeight,
      )}>
      <QuestionAnswerPicker question={question} interactive={false} />
    </div>
    <ProgressBar countdown={countdown} />
  </Page>
)

export default HostQuestionState
