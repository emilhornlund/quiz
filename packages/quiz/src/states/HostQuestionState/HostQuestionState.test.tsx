import { GameEventType, QuestionType } from '@quiz/common'
import { render } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import HostQuestionState from './HostQuestionState'

describe('HostQuestionState', () => {
  it('should render HostQuestionState with question type multi-choice and two answers', async () => {
    const { container } = render(
      <HostQuestionState
        event={{
          type: GameEventType.GameQuestionHost,
          game: {
            pin: '123456',
          },
          question: {
            type: QuestionType.MultiChoice,
            question: 'Who painted The Starry Night?',
            imageURL:
              'https://i.pinimg.com/originals/a6/60/72/a66072b0e88258f2898a76c3f3c01041.jpg',
            answers: [
              { value: 'Vincent van Gogh' },
              { value: 'Pablo Picasso' },
            ],
            duration: 30,
          },
          countdown: {
            expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
            serverTime: new Date().toISOString(),
          },
          submissions: { current: 3, total: 10 },
          pagination: { current: 1, total: 20 },
        }}
      />,
    )

    expect(container).toMatchSnapshot()
  })

  it('should render HostQuestionState with question type multi-choice and four answers', async () => {
    const { container } = render(
      <HostQuestionState
        event={{
          type: GameEventType.GameQuestionHost,
          game: {
            pin: '123456',
          },
          question: {
            type: QuestionType.MultiChoice,
            question: 'Who painted The Starry Night?',
            imageURL: 'https://wallpapercave.com/wp/wp2824407.jpg',
            answers: [
              { value: 'Vincent van Gogh' },
              { value: 'Pablo Picasso' },
              { value: 'Leonardo da Vinci' },
              { value: 'Claude Monet' },
            ],
            duration: 30,
          },
          countdown: {
            expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
            serverTime: new Date().toISOString(),
          },
          submissions: { current: 3, total: 10 },
          pagination: { current: 1, total: 20 },
        }}
      />,
    )

    expect(container).toMatchSnapshot()
  })

  it('should render HostQuestionState with question type multi-choice and six answers', async () => {
    const { container } = render(
      <HostQuestionState
        event={{
          type: GameEventType.GameQuestionHost,
          game: {
            pin: '123456',
          },
          question: {
            type: QuestionType.MultiChoice,
            question: 'Who painted The Starry Night?',
            imageURL: 'https://wallpaperaccess.com/full/157316.jpg',
            answers: [
              { value: 'Vincent van Gogh' },
              { value: 'Pablo Picasso' },
              { value: 'Leonardo da Vinci' },
              { value: 'Claude Monet' },
              { value: 'Michelangelo' },
              { value: 'Rembrandt' },
            ],
            duration: 30,
          },
          countdown: {
            expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
            serverTime: new Date().toISOString(),
          },
          submissions: { current: 3, total: 10 },
          pagination: { current: 1, total: 20 },
        }}
      />,
    )

    expect(container).toMatchSnapshot()
  })

  it('should render HostQuestionState with question type range', async () => {
    const { container } = render(
      <HostQuestionState
        event={{
          type: GameEventType.GameQuestionHost,
          game: {
            pin: '123456',
          },
          question: {
            type: QuestionType.Range,
            question:
              "What percentage of the earth's surface is covered by water?",
            min: 0,
            max: 100,
            step: 1,
            duration: 30,
          },
          countdown: {
            expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
            serverTime: new Date().toISOString(),
          },
          submissions: { current: 3, total: 10 },
          pagination: { current: 1, total: 20 },
        }}
      />,
    )

    expect(container).toMatchSnapshot()
  })

  it('should render HostQuestionState with question type true false', async () => {
    const { container } = render(
      <HostQuestionState
        event={{
          type: GameEventType.GameQuestionHost,
          game: {
            pin: '123456',
          },
          question: {
            type: QuestionType.TrueFalse,
            question: "Rabbits can't vomit?",
            duration: 30,
          },
          countdown: {
            expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
            serverTime: new Date().toISOString(),
          },
          submissions: { current: 3, total: 10 },
          pagination: { current: 1, total: 20 },
        }}
      />,
    )

    expect(container).toMatchSnapshot()
  })

  it('should render HostQuestionState with question type type answer', async () => {
    const { container } = render(
      <HostQuestionState
        event={{
          type: GameEventType.GameQuestionHost,
          game: {
            pin: '123456',
          },
          question: {
            type: QuestionType.TypeAnswer,
            question: 'Who painted the Mono Lisa?',
            duration: 30,
          },
          countdown: {
            expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
            serverTime: new Date().toISOString(),
          },
          submissions: { current: 3, total: 10 },
          pagination: { current: 1, total: 20 },
        }}
      />,
    )

    expect(container).toMatchSnapshot()
  })
})
