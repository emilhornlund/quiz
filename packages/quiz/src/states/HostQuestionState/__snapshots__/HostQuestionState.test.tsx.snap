// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HostQuestionState > should render HostQuestionState with question type multi and four answers 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      >
        <div
          class="main small"
        >
          <button
            class="secondary"
            data-testid="test-skip-button-button"
            id="skip-button"
            name="skip-button"
            type="button"
          >
            Skip
            <div
              class="arrow right"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="title large"
            >
              Who painted The Starry Night?
            </div>
          </div>
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-hourglass "
                data-icon="hourglass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 32C0 14.3 14.3 0 32 0L64 0 320 0l32 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l0 11c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1l0 11c17.7 0 32 14.3 32 32s-14.3 32-32 32l-32 0L64 512l-32 0c-17.7 0-32-14.3-32-32s14.3-32 32-32l0-11c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75l0-11C14.3 64 0 49.7 0 32zM96 64l0 11c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9l0-11L96 64zm0 384l192 0 0-11c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9l0 11z"
                  fill="currentColor"
                />
              </svg>
              <span>
                30
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-comment "
                data-icon="comment"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 240c0 114.9-114.6 208-256 208c-37.1 0-72.3-6.4-104.1-17.9c-11.9 8.7-31.3 20.6-54.3 30.6C73.6 471.1 44.7 480 16 480c-6.5 0-12.3-3.9-14.8-9.9c-2.5-6-1.1-12.8 3.4-17.4c0 0 0 0 0 0s0 0 0 0s0 0 0 0c0 0 0 0 0 0l.3-.3c.3-.3 .7-.7 1.3-1.4c1.1-1.2 2.8-3.1 4.9-5.7c4.1-5 9.6-12.4 15.2-21.6c10-16.6 19.5-38.4 21.4-62.9C17.7 326.8 0 285.1 0 240C0 125.1 114.6 32 256 32s256 93.1 256 208z"
                  fill="currentColor"
                />
              </svg>
              <span>
                3
                 / 
                10
              </span>
            </div>
          </div>
        </div>
        <div
          class="row fullHeight"
        >
          <div
            class="column largeWidth"
          >
            <img
              alt="Who painted The Starry Night?"
              class="image"
              src="https://wallpapercave.com/wp/wp2824407.jpg"
            />
          </div>
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            WIP component for: 
            MULTI
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          1
           / 
          20
        </div>
        <div
          class="gamePIN"
        >
          Game PIN: 
          <span>
            123456
          </span>
        </div>
        <div
          class="actions"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`HostQuestionState > should render HostQuestionState with question type multi and six answers 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      >
        <div
          class="main small"
        >
          <button
            class="secondary"
            data-testid="test-skip-button-button"
            id="skip-button"
            name="skip-button"
            type="button"
          >
            Skip
            <div
              class="arrow right"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="title large"
            >
              Who painted The Starry Night?
            </div>
          </div>
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-hourglass "
                data-icon="hourglass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 32C0 14.3 14.3 0 32 0L64 0 320 0l32 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l0 11c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1l0 11c17.7 0 32 14.3 32 32s-14.3 32-32 32l-32 0L64 512l-32 0c-17.7 0-32-14.3-32-32s14.3-32 32-32l0-11c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75l0-11C14.3 64 0 49.7 0 32zM96 64l0 11c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9l0-11L96 64zm0 384l192 0 0-11c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9l0 11z"
                  fill="currentColor"
                />
              </svg>
              <span>
                30
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-comment "
                data-icon="comment"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 240c0 114.9-114.6 208-256 208c-37.1 0-72.3-6.4-104.1-17.9c-11.9 8.7-31.3 20.6-54.3 30.6C73.6 471.1 44.7 480 16 480c-6.5 0-12.3-3.9-14.8-9.9c-2.5-6-1.1-12.8 3.4-17.4c0 0 0 0 0 0s0 0 0 0s0 0 0 0c0 0 0 0 0 0l.3-.3c.3-.3 .7-.7 1.3-1.4c1.1-1.2 2.8-3.1 4.9-5.7c4.1-5 9.6-12.4 15.2-21.6c10-16.6 19.5-38.4 21.4-62.9C17.7 326.8 0 285.1 0 240C0 125.1 114.6 32 256 32s256 93.1 256 208z"
                  fill="currentColor"
                />
              </svg>
              <span>
                3
                 / 
                10
              </span>
            </div>
          </div>
        </div>
        <div
          class="row fullHeight"
        >
          <div
            class="column largeWidth"
          >
            <img
              alt="Who painted The Starry Night?"
              class="image"
              src="https://wallpaperaccess.com/full/157316.jpg"
            />
          </div>
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            WIP component for: 
            MULTI
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          1
           / 
          20
        </div>
        <div
          class="gamePIN"
        >
          Game PIN: 
          <span>
            123456
          </span>
        </div>
        <div
          class="actions"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`HostQuestionState > should render HostQuestionState with question type multi and two answers 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      >
        <div
          class="main small"
        >
          <button
            class="secondary"
            data-testid="test-skip-button-button"
            id="skip-button"
            name="skip-button"
            type="button"
          >
            Skip
            <div
              class="arrow right"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="title large"
            >
              Who painted The Starry Night?
            </div>
          </div>
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-hourglass "
                data-icon="hourglass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 32C0 14.3 14.3 0 32 0L64 0 320 0l32 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l0 11c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1l0 11c17.7 0 32 14.3 32 32s-14.3 32-32 32l-32 0L64 512l-32 0c-17.7 0-32-14.3-32-32s14.3-32 32-32l0-11c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75l0-11C14.3 64 0 49.7 0 32zM96 64l0 11c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9l0-11L96 64zm0 384l192 0 0-11c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9l0 11z"
                  fill="currentColor"
                />
              </svg>
              <span>
                30
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-comment "
                data-icon="comment"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 240c0 114.9-114.6 208-256 208c-37.1 0-72.3-6.4-104.1-17.9c-11.9 8.7-31.3 20.6-54.3 30.6C73.6 471.1 44.7 480 16 480c-6.5 0-12.3-3.9-14.8-9.9c-2.5-6-1.1-12.8 3.4-17.4c0 0 0 0 0 0s0 0 0 0s0 0 0 0c0 0 0 0 0 0l.3-.3c.3-.3 .7-.7 1.3-1.4c1.1-1.2 2.8-3.1 4.9-5.7c4.1-5 9.6-12.4 15.2-21.6c10-16.6 19.5-38.4 21.4-62.9C17.7 326.8 0 285.1 0 240C0 125.1 114.6 32 256 32s256 93.1 256 208z"
                  fill="currentColor"
                />
              </svg>
              <span>
                3
                 / 
                10
              </span>
            </div>
          </div>
        </div>
        <div
          class="row fullHeight"
        >
          <div
            class="column largeWidth"
          >
            <img
              alt="Who painted The Starry Night?"
              class="image"
              src="https://i.pinimg.com/originals/a6/60/72/a66072b0e88258f2898a76c3f3c01041.jpg"
            />
          </div>
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            WIP component for: 
            MULTI
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          1
           / 
          20
        </div>
        <div
          class="gamePIN"
        >
          Game PIN: 
          <span>
            123456
          </span>
        </div>
        <div
          class="actions"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`HostQuestionState > should render HostQuestionState with question type slider 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      >
        <div
          class="main small"
        >
          <button
            class="secondary"
            data-testid="test-skip-button-button"
            id="skip-button"
            name="skip-button"
            type="button"
          >
            Skip
            <div
              class="arrow right"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="title large"
            >
              What percentage of the earth's surface is covered by water?
            </div>
          </div>
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-hourglass "
                data-icon="hourglass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 32C0 14.3 14.3 0 32 0L64 0 320 0l32 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l0 11c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1l0 11c17.7 0 32 14.3 32 32s-14.3 32-32 32l-32 0L64 512l-32 0c-17.7 0-32-14.3-32-32s14.3-32 32-32l0-11c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75l0-11C14.3 64 0 49.7 0 32zM96 64l0 11c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9l0-11L96 64zm0 384l192 0 0-11c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9l0 11z"
                  fill="currentColor"
                />
              </svg>
              <span>
                30
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-comment "
                data-icon="comment"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 240c0 114.9-114.6 208-256 208c-37.1 0-72.3-6.4-104.1-17.9c-11.9 8.7-31.3 20.6-54.3 30.6C73.6 471.1 44.7 480 16 480c-6.5 0-12.3-3.9-14.8-9.9c-2.5-6-1.1-12.8 3.4-17.4c0 0 0 0 0 0s0 0 0 0s0 0 0 0c0 0 0 0 0 0l.3-.3c.3-.3 .7-.7 1.3-1.4c1.1-1.2 2.8-3.1 4.9-5.7c4.1-5 9.6-12.4 15.2-21.6c10-16.6 19.5-38.4 21.4-62.9C17.7 326.8 0 285.1 0 240C0 125.1 114.6 32 256 32s256 93.1 256 208z"
                  fill="currentColor"
                />
              </svg>
              <span>
                3
                 / 
                10
              </span>
            </div>
          </div>
        </div>
        <div
          class="row fullHeight"
        >
          <div
            class="column largeWidth"
          />
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            WIP component for: 
            SLIDER
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          1
           / 
          20
        </div>
        <div
          class="gamePIN"
        >
          Game PIN: 
          <span>
            123456
          </span>
        </div>
        <div
          class="actions"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`HostQuestionState > should render HostQuestionState with question type true false 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      >
        <div
          class="main small"
        >
          <button
            class="secondary"
            data-testid="test-skip-button-button"
            id="skip-button"
            name="skip-button"
            type="button"
          >
            Skip
            <div
              class="arrow right"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="title large"
            >
              Rabbits can't vomit?
            </div>
          </div>
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-hourglass "
                data-icon="hourglass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 32C0 14.3 14.3 0 32 0L64 0 320 0l32 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l0 11c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1l0 11c17.7 0 32 14.3 32 32s-14.3 32-32 32l-32 0L64 512l-32 0c-17.7 0-32-14.3-32-32s14.3-32 32-32l0-11c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75l0-11C14.3 64 0 49.7 0 32zM96 64l0 11c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9l0-11L96 64zm0 384l192 0 0-11c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9l0 11z"
                  fill="currentColor"
                />
              </svg>
              <span>
                30
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-comment "
                data-icon="comment"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 240c0 114.9-114.6 208-256 208c-37.1 0-72.3-6.4-104.1-17.9c-11.9 8.7-31.3 20.6-54.3 30.6C73.6 471.1 44.7 480 16 480c-6.5 0-12.3-3.9-14.8-9.9c-2.5-6-1.1-12.8 3.4-17.4c0 0 0 0 0 0s0 0 0 0s0 0 0 0c0 0 0 0 0 0l.3-.3c.3-.3 .7-.7 1.3-1.4c1.1-1.2 2.8-3.1 4.9-5.7c4.1-5 9.6-12.4 15.2-21.6c10-16.6 19.5-38.4 21.4-62.9C17.7 326.8 0 285.1 0 240C0 125.1 114.6 32 256 32s256 93.1 256 208z"
                  fill="currentColor"
                />
              </svg>
              <span>
                3
                 / 
                10
              </span>
            </div>
          </div>
        </div>
        <div
          class="row fullHeight"
        >
          <div
            class="column largeWidth"
          />
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            WIP component for: 
            TRUE_FALSE
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          1
           / 
          20
        </div>
        <div
          class="gamePIN"
        >
          Game PIN: 
          <span>
            123456
          </span>
        </div>
        <div
          class="actions"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`HostQuestionState > should render HostQuestionState with question type type answer 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </div>
      <div
        class="side"
      >
        <div
          class="main small"
        >
          <button
            class="secondary"
            data-testid="test-skip-button-button"
            id="skip-button"
            name="skip-button"
            type="button"
          >
            Skip
            <div
              class="arrow right"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="content center"
    >
      <div
        class="main"
      >
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="title large"
            >
              Who painted the Mono Lisa?
            </div>
          </div>
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-hourglass "
                data-icon="hourglass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 32C0 14.3 14.3 0 32 0L64 0 320 0l32 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l0 11c0 42.4-16.9 83.1-46.9 113.1L237.3 256l67.9 67.9c30 30 46.9 70.7 46.9 113.1l0 11c17.7 0 32 14.3 32 32s-14.3 32-32 32l-32 0L64 512l-32 0c-17.7 0-32-14.3-32-32s14.3-32 32-32l0-11c0-42.4 16.9-83.1 46.9-113.1L146.7 256 78.9 188.1C48.9 158.1 32 117.4 32 75l0-11C14.3 64 0 49.7 0 32zM96 64l0 11c0 25.5 10.1 49.9 28.1 67.9L192 210.7l67.9-67.9c18-18 28.1-42.4 28.1-67.9l0-11L96 64zm0 384l192 0 0-11c0-25.5-10.1-49.9-28.1-67.9L192 301.3l-67.9 67.9c-18 18-28.1 42.4-28.1 67.9l0 11z"
                  fill="currentColor"
                />
              </svg>
              <span>
                30
              </span>
            </div>
          </div>
          <div
            class="column"
          >
            <div
              class="iconInfo"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-comment "
                data-icon="comment"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 240c0 114.9-114.6 208-256 208c-37.1 0-72.3-6.4-104.1-17.9c-11.9 8.7-31.3 20.6-54.3 30.6C73.6 471.1 44.7 480 16 480c-6.5 0-12.3-3.9-14.8-9.9c-2.5-6-1.1-12.8 3.4-17.4c0 0 0 0 0 0s0 0 0 0s0 0 0 0c0 0 0 0 0 0l.3-.3c.3-.3 .7-.7 1.3-1.4c1.1-1.2 2.8-3.1 4.9-5.7c4.1-5 9.6-12.4 15.2-21.6c10-16.6 19.5-38.4 21.4-62.9C17.7 326.8 0 285.1 0 240C0 125.1 114.6 32 256 32s256 93.1 256 208z"
                  fill="currentColor"
                />
              </svg>
              <span>
                3
                 / 
                10
              </span>
            </div>
          </div>
        </div>
        <div
          class="row fullHeight"
        >
          <div
            class="column largeWidth"
          />
        </div>
        <div
          class="row flexibleHeight"
        >
          <div
            class="column"
          >
            WIP component for: 
            TYPE_ANSWER
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          1
           / 
          20
        </div>
        <div
          class="gamePIN"
        >
          Game PIN: 
          <span>
            123456
          </span>
        </div>
        <div
          class="actions"
        />
      </div>
    </div>
  </div>
</div>
`;
